docs/readme.md

```diff
@@ -0,0 +1,76 @@
+# Narwhal Extensions for Angular
```

Comment:
This is the steps I follow when I explain Nx to our clients:



Large companies specific requirements when it comes to building software.

1. They don't build small apps. They have multiple teams building multiple apps using multiple shared libs. It's many to many to many. Organizing this dev workflow is challenging.
2. They care about consistency. AngularCLI has some conventions but they don't go far enough. What about state management, what about routing? Even small things code formatting?
3. They have legacy angularjs apps they need to migrate. NgUpgrade is great, but still a lot of work to set up.
4. They want to write robust code: error handling, race conditions.
5. ...



These are, among other, problems that Nx addresses.

1. It gives you Nx Workspace--a monorepo way of building frontend apps. The same approach used by Google, Facebook, Uber, Twitter and many others.
2. It gives you code generation abilities to generate state management stuff and routing. It also comes with runtime libraries to make your code more robust. It comes with a code formatter (used by Google!).
3. It comes with code generation abilities to set up a hybrid app in minutes. So you can take advantage of your existing angularjs code right away.
...



Then I talk about each of those parts separately. And I cover the two node modules separately.

Basically I try to explain the problems people have first, before going into what schematics are. What do you think?

Reply:
Added a "why Nx" section with this info.